warehouse/ingest-core/src/main/java/datawave/ingest/mapreduce/job/reindex/ShardReindexMapper.java

```diff
@@ -0,0 +1,764 @@
+        final int fiBaseLength = FI_START_BYTES.length;
+        if (cf.length <= fiBaseLength || WritableComparator.compareBytes(cf, 0, fiBaseLength, FI_START_BYTES, 0, fiBaseLength) != 0) {
+            // increment count of non-fi key
+            incrementCounter("key types", "non-fi");
+            if (cf.length == 2) {
+                // tf is the only shard Key of length 2
+                incrementCounter("key types", "tf");
+
+                // get the tf field
+                final String tfField = getFieldFromTF(key);
+
+                // if reprocessing events and exporting shard and either not generating tf or this is an index only field write it to the context
+                if (this.reprocessEvents && this.exportShard && (!this.generateTF || this.defaultHelper.isIndexOnlyField(tfField))) {
+                    context.write(new BulkIngestKey(shardTable, key), value);
+                    incrementCounter("tf", tfField);
+                }
+
+                context.progress();
+                // nothing else to do for this Key
+                return;
+            }
+
+            // check for d column
+            if (cf.length == 1 && cf[0] == 'd') {
+                incrementCounter("key types", "d");
+                if (this.reprocessEvents && this.exportShard) {
+                    context.write(new BulkIngestKey(shardTable, key), value);
+                    incrementCounter("export", "d");
+                }
+                context.progress();
+                return;
+            }
+
+            // everything else should be an event
+            if (this.reprocessEvents) {
+                try {
+                    processEvent(context, key);
+                    if (exportShard) {
+                        context.write(new BulkIngestKey(shardTable, key), value);
+                        incrementCounter("export", "event");
+                    }
+                } catch (ClassNotFoundException | InvocationTargetException | InstantiationException | IllegalAccessException | NoSuchMethodException e) {
+                    throw new RuntimeException("Could not process event for: " + key, e);
+                }
+            }
+            context.progress();
+            return;
+        }
+
+        // if none of the key types above, Key is an fi Key type
+        processFI(context, key);
+        context.progress();
+    }
+
+    // TODO add javadoc
+    private void processFI(Context context, Key key) throws IOException, InterruptedException {
+        final byte[] cf = key.getColumnFamilyData().getBackingArray();
+
+        // check if it's the same target field as the last one
+        final int fiBaseLength = FI_START_BYTES.length;
+        final int fiBaseOffset = fiBaseLength + 1;
+        // TODO add comment here
+        if (this.lastFiBytes == null || WritableComparator.compareBytes(cf, fiBaseOffset, cf.length - fiBaseOffset, this.lastFiBytes, fiBaseOffset,
```

Comment:
Rather than commenting, you could turn this into a method call:

`if (isNewFiField(cf))`